lib/elaborate/elaborateReferences.ts

```diff
@@ -346,10 +335,11 @@ export class ElaborateNames {
     if (clearVisibilityMap) {
       this.scopeVisibilityMap.clear();
     }
+    this.scopeVisibilityMap.delete(parent);
```

Comment:
why do you always delete the parent?

Reply:
Because it does not work otherwise

Reply:
But now changed it to fully clear the map after useClauses

Reply:
But this is what the `clearVisibilityMap` boolean is for. Either
- fix the use of this boolean in `elaborateUseClauses` and remove `elaborateNames.ts:28` or,
- remove the `clearVisibilityMap` boolean in `elaborateUseClauses`